File: JAVA/2319. Check if Matrix Is X-Matrix.java
class Solution {
    public boolean checkXMatrix(int[][] g) {
        int i,j;
        for(i=0;i<g.length;i++)
        {
            for(j=0;j<g.length;j++)
            {
                if(i==j || i+j+1==g.length){
                    if(g[i][j]==0)
                    return false;
                }
                else{
                    if(g[i][j]!=0)
                    return false;
                }
            }
        }
        return true;
    }
}

File: C++/2319. Check if Matrix Is X-Matrix.cpp
class Solution {
public:
    bool checkXMatrix(vector<vector<int>>& g) {
        int i, j, n = g.size();
        for(i = 0; i < n; i++){
            for(j = 0; j < n; j++){
                if(i == j || i + j == n - 1) {
                    if(g[i][j] == 0) 
                        return false;
                } else { 
                    if(g[i][j] != 0) 
                        return false;
                }
            }
        }
        return true;
    }
};
